spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/zookeeper/ZooKeeperStreamRepository.java

```diff
@@ -282,7 +282,7 @@ public void deleteAll() {
 
 		List<Stream> results = new ArrayList<Stream>();
 		for (Stream stream : all) {
-			if (stream.getDefinition().getName().compareTo(to) > 1) {
+			if (stream.getDefinition().getName().compareTo(to) > 0) {
```

Comment:
That whole class needs to be revisited. That's why I created: https://jira.spring.io/browse/XD-2620
For instance the class is untested, but it looks like adding test at this point seems tricky.